src/query/graphite/storage/m3_wrapper.go

```diff
@@ -141,19 +141,28 @@ func truncateBoundsToResolution(
 
 func translateTimeseries(
 	ctx xctx.Context,
-	result *storage.FetchResult,
+	result block.Result,
 	start, end time.Time,
 ) ([]*ts.Series, error) {
-	m3list := result.SeriesList
-	series := make([]*ts.Series, len(m3list))
+	block := result.Blocks[0]
+	defer block.Close()
+
+	iter, err := block.SeriesIter()
+	if err != nil {
+		return nil, err
+	}
+
+	seriesMetas := iter.SeriesMeta()
 	resolutions := result.Metadata.Resolutions
-	if len(series) != len(resolutions) {
+	if len(seriesMetas) != len(resolutions) {
 		return nil, fmt.Errorf("number of timeseries %d does not match number of "+
-			"resolutions %d", len(series), len(resolutions))
+			"resolutions %d", len(seriesMetas), len(resolutions))
 	}
 
-	for i, m3series := range m3list {
-		resolution := time.Duration(resolutions[i])
+	idx := 0
```

Comment:
nit: Would it be better to make this part of the for loop? `for idx := 0; iter.Next(); idx++ { ... }`

Reply:
Hm, always seems a bit weird to me when the loop incrementing don't correspond to the conditional, but I suppose it may be a bit safer if a new path is added that `continue`s or something like that